conversion/conversion.go

```diff
@@ -67,25 +67,63 @@ import (
 	dydb "github.com/aws/aws-sdk-go/service/dynamodb"
 	"github.com/aws/aws-sdk-go/service/dynamodbstreams"
 	mysqldriver "github.com/go-sql-driver/mysql"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/stdlib"
 	"go.uber.org/zap"
 	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
 	"google.golang.org/grpc/metadata"
 	"google.golang.org/protobuf/proto"
-	"github.com/jackc/pgx/v5"
-    "github.com/jackc/pgx/v5/stdlib"
 )
 
 var (
 	// Set the maximum number of concurrent workers during foreign key creation.
 	// This number should not be too high so as to not hit the AdminQuota limit.
 	// AdminQuota limits are mentioned here: https://cloud.google.com/spanner/quotas#administrative_limits
 	// If facing a quota limit error, consider reducing this value.
-	MaxWorkers       = 50
-	once             sync.Once
-	datastreamClient *datastream.Client
+	MaxWorkers          = 50
+	once                sync.Once
+	datastreamClient    *datastream.Client
+	resourcesForCleanup []*ConnectionProfileReq
 )
 
-func getDatastreamClient(ctx context.Context) *datastream.Client {
+type ConnectionProfile struct {
+	// Project Id of the resource
+	ProjectId string
+	// Datashard Id for the resource
+	DatashardId string
+	// Name of connection profile
+	Id string
+	// If true, don't create resource, only validate if creation is possible. If false, create resource.
+	ValidateOnly bool
+	// If true, create source connection profile, else create target connection profile and gcs bucket.
+	IsSource bool
+	// For source connection profile host of MySql instance
+	Host string
+	// For source connection profile port of MySql instance
+	Port string
+	// For source connection profile password of MySql instance
+	Password string
+	// For source connection profile user name of MySql instance
+	User string
+	// Region of connection profile to be created
+	Region string
+	// For target connection profile name of gcs bucket to be created
+	BucketName string
+}
```

Comment:
I think a similar struct exists elsewhere, can you check if there is an opportunity to reuse?

Reply:
We do, but it is in UI and has some missing fields. That would get removed when the two sharded and non-sharded flows will merge